docs/deprecation_policy.md

```diff
@@ -0,0 +1,37 @@
+Deprecation Policy for Platform Dependencies
+============================================
+
+Synapse has a number of platform dependencies, including Python and PostgreSQL.
+This document outlines the policy towards which versions we support, and how we
+drop support for versions in the future.
+
+
+Policy
+------
+
+For Python and PostgreSQL Synapse will follow the upstream support life cycles,
+i.e. when a version reaches End of Life Synapse will withdraw support for that
+version in future releases.
+
+The team will not intentionally break compatibility for at least one month after
+an EOL date, but does explicitly reserve the right to do so should security or
+other concerns arise in an unmaintained version of a dependency.
+
+
+Details on the upstream support life cycles for Python and PostgreSQL are
+documented at https://endoflife.date/python and
+https://endoflife.date/postgresql.
+
```

Comment:
Do we not want to comment on the policy for any other dependencies?

Reply:
Do we have other deps we care about in this regard?

Reply:
Probably not, but it may be nice to explicitly say that we don't.

Reply:
We can always add more in the future. 🤷